org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java

```diff
@@ -267,17 +268,42 @@ public static Location toLocation(IJavaElement element) throws JavaModelExceptio
 			return null;
 		}
 		if (element instanceof ISourceReference) {
-			ISourceRange nameRange = ((ISourceReference) element).getNameRange();
-			if(cf == null){
-				return toLocation(unit,nameRange.getOffset(), nameRange.getLength());
-			}else{
-				return toLocation(cf,nameRange.getOffset(), nameRange.getLength());
+			ISourceRange nameRange = getNameRange(element);
+			if (nameRange != null && SourceRange.isAvailable(nameRange)) {
+				if (cf == null) {
+					return toLocation(unit, nameRange.getOffset(), nameRange.getLength());
+				} else {
+					return toLocation(cf, nameRange.getOffset(), nameRange.getLength());
+				}
 			}
 		}
 		return null;
 	}
 
 	/**

```

Comment:
SourceRange.isAvailable(nameRange) already checks for null. No need to do a null check before calling this method.